src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -375,9 +375,18 @@ private static void LoadSourceFiles(CanvasDocument app, DirectoryReader director
             }
 
             // When loading TestSuites sharded files, add them within the top parent AppTest control (i.e. Test_7F478737223C4B69)
+            // Make sure to load the the Test_7F478737223C4B69.fx.yaml file first to add the top parent control.
+            var appTestControl = directory.EnumerateFiles(TestDir, AppTestControlName + ".fx.yaml").FirstOrDefault();
+            if (appTestControl != null)
+            {
+                AddControl(app, appTestControl._relativeName, false, appTestControl.GetContents(), errors);
+            }
             foreach (var file in directory.EnumerateFiles(TestDir, "*.fx.yaml"))
             {
-                AddControl(app, file._relativeName, false, file.GetContents(), errors);
+                if (file._relativeName != AppTestControlName + ".fx.yaml")
```

Comment:
possible casing bug (if filenames are case insensitive)?

Reply:
This the top parent control for AppTest which has fixed name "Test_7F478737223C4B69"

Reply:
My concern is that the file system is case insensitive, and so could it somehow loses case  (becomes "TEST_7F..."). Basically, should this be a string insensitive compare? Or should we have a higher level helper for comparing filenames? (our "FileKind" sort of handles that)

Reply:
updated the code to use FileKind